WalletWasabi/Blockchain/Analysis/BlockchainAnalyzer.cs

```diff
@@ -64,12 +64,14 @@ public void Analyze(SmartTransaction tx)
 
 	private static void AnalyzeCoinjoinWalletInputs(SmartTransaction tx, out double mixedAnonScore, out double nonMixedAnonScore)
 	{
+		CoinjoinAnalyzer cjAnal = new(tx);
+
 		// Consolidation in coinjoins is the only type of consolidation that's acceptable,
 		// because coinjoins are an exception from common input ownership heuristic.
 		// Calculate weighted average.
-		mixedAnonScore = tx.WalletInputs.Sum(x => x.HdPubKey.AnonymitySet * x.Amount.Satoshi) / tx.WalletInputs.Sum(x => x.Amount);
+		mixedAnonScore = tx.WalletVirtualInputs.Sum(x => (Math.Max(1, x.HdPubKey.AnonymitySet - cjAnal.ComputeInputSanction(x))) * x.Amount.Satoshi) / tx.WalletVirtualInputs.Sum(x => x.Amount);
 
-		nonMixedAnonScore = tx.WalletInputs.Min(x => x.HdPubKey.AnonymitySet);
+		nonMixedAnonScore = tx.WalletVirtualInputs.Min(x => Math.Max(1, x.HdPubKey.AnonymitySet - cjAnal.ComputeInputSanction(x)));
```

Comment:
I think the usage of this is just wrong. For example nonMixedAnonScore shouldn't go lower than min wallet inputs' anon score. It makes no sense to start out lower than that.